### ISSUE TITLE
refresh qrcode

### ISSUE THREAD

**Issue by MLiranG** [adiwajshing/Baileys#38]
Maybe create a on-qrcode event? or there is already an option for qr-code refreshment

**Comment by adiwajshing** [comment]
Well, if you want to handle the QR code generation, you can override the following method:

``` js
client.onReadyForPhoneAuthentication = ([ref, publicKey, clientID]) => {
    const str = ref + "," + publicKey + "," + clientID
    // generate QR with str
}
```

You can find this method [here](https://github.com/adiwajshing/Baileys/blob/1ab251d8ccd62e524111ddd86df079066896c57b/WhatsAppWeb.js#L92).

**Comment by MLiranG** [comment]
> ```js
> client.onReadyForPhoneAuthentication = ([ref, publicKey, clientID]) => {
>     const str = ref + "," + publicKey + "," + clientID
>     // generate QR with str
> }
> ```

oh ok
idk if this help is welcomed here but thats my code and thats my console, why does the process exit instead of console.log another qrcode?

```js
const WhatsAppWeb = require('baileys')
const client = new WhatsAppWeb() 
client.logUnhandledMessages = true
client.onReadyForPhoneAuthentication = ([ref, publicKey, clientID]) => {
    const str = ref + "," + publicKey + "," + clientID
    console.log(str)
}
client.connect()
.then (([user, chats, contacts, unread]) => {
    console.log ("oh hello " + user.name + " (" + user.id + ")")
    console.log ("you have " + unread.length + " unread messages")
    console.log ("you have " + chats.length + " chats")
})
.catch (err => console.log("unexpected error: " + err) )
```

![image](https://user-images.githubusercontent.com/30736093/85208921-aff99300-b33c-11ea-8845-10c91a4f2cf1.png)

**Comment by adiwajshing** [comment]
That should not happen. Can you add the following callback and see if something happens:

``` javascript
client.setOnUnexpectedDisconnect (err => console.log (err))
```

**Comment by MLiranG** [comment]
Yes I did, it didn't print nothing

**Comment by adiwajshing** [comment]
Okay. Are you running it with pure node or some other tool?

**Comment by MLiranG** [comment]
pure node( if nodemon is pure node either )

**Comment by MLiranG** [comment]
yeah, tried it with `node ./main.js` either and it still doesn't work

**Comment by adiwajshing** [comment]
So, I just tested it out and its working just fine. Could you try using the master branch in your project instead of using the npm package?

You can do so with: `npm i github:adiwajshing/Baileys`

**Comment by MLiranG** [comment]
Oh ok, ill try now

‫בתאריך יום א׳, 21 ביוני 2020 ב-12:34 מאת ‪Adhiraj Singh‬‏ <‪
notifications@github.com‬‏>:‬

**Comment by MLiranG** [comment]
it still does the same thing, `[nodemon] clean exit - waiting for changes before restart`

**Comment by adiwajshing** [comment]
@MLiranG can you please check if the issue is fixed in version 2.0.0? Thanks.

**Comment by adiwajshing** [comment]
Assuming it's cool now. Closing.
